test-vectors/schema.json

```diff
@@ -0,0 +1,156 @@
+{
+  "$id": "https://filecoin.io/oni/schemas/test-vector.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "a filecoin VM test vector",
+  "type": "object",
+  "definitions": {
+    "hex": {
+      "title": "hex value",
+      "description": "a hex value prefixed with 0x, and accepting only lowercase characters; 0x represents an empty byte array",
+      "type": "string",
+      "pattern": "0x[0-9a-f]*",
+      "examples": [
+        "0xa1b2c3",
+        "0x"
+      ]
+    },
+    "meta": {
+      "title": "metadata",
+      "description": "metadata about this test vector, such as its id, version, data about its generation, etc.",
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "id": {
+          "type": "string"
+        },
+        "version": {
+          "type": "string"
+        },
+        "gen": {
+          "title": "generation metadata",
+          "description": "metadata about how this test vector was generated",
+          "type": "object",
+          "additionalProperties": false,
+          "properties": {
+            "source": {
+              "type": "string",
+              "examples": [
+                "lotus",
+                "dsl"
+              ]
+            },
+            "version": {
+              "type": "string",
+              "examples": [
+                "0.4.1+git.27d74337+api0.8.1"
+              ]
+            }
+          }
+        }
+      }
+    },
+    "preconditions": {
+      "title": "execution preconditions",
+      "description": "preconditions that need to be applied and satisfied before this test vector can be executed",
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "state_tree": {
+          "title": "state tree to seed",
+          "description": "state tree to seed before applying this test vector; mapping of actor addresses => serialized state",
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/hex"
+          },
+          "examples": [
+            {
+              "t01": "0x0123456789abcdef",
+              "t02": "0x0123456789abcdef",
+              "t03": "0x0123456789abcdef"
```

Comment:
Do you see these serializations eventually being expanded out into a schema we can use to author tests solely in JSON?

Reply:
From what I understand we are not supposed to do that - see https://docs.google.com/document/d/1_wlRp-DBVqlTijP-Jq3SzSVd65ABgnXD1pw_yBxip3I/edit?disco=AAAAKA9s15M

Reply:
we should use the same definition / schema format for state_tree in preconditions and post conditions. It's extra work to store it as an array of actor states as the pre-condition, and then as a car + root in the post condition.